### ISSUE TITLE
Invalid magics in dask.imperitive.Value

### ISSUE THREAD

**Issue by llllllllll** [ContinuumIO/dask#462]
```python
__inv__
__concat__
__not__
```

You are setting these to `operator.inv`, `operator.concat` and `operator.not_`; however, `inv` is an alias for `invert` and the magic is already defined. `concat` is an alias for `add` that is more readable for lists and collections. Also, `not_` cannot be overridden with a function, `__not__` is not called.

If you are using these names internally then I apologize; however, I would urge you not to "invent" your own magic methods:

from pep8:
> `__double_leading_and_trailing_underscore__` : "magic" objects or attributes that live in user-controlled namespaces. E.g. `__init__ `, `__import__` or `__file__` . Never invent such names; only use them as documented.

**Comment by jcrist** [comment]
I literally just copied all of these that seemed valid over from [the operator docs](https://docs.python.org/2/library/operator.html), but definitely haven't try all of them. You are correct about all 3, will fix. Thanks!

**Comment by llllllllll** [comment]
Sorry I didn't catch the rest sooner; however, I just looked back at my last python implementation of thunk and found some more magics. These might not all make sense, but at least the math ones sort of do.

```
__round__
__floor__
__ceil__
__trunc__
__instancecheck__
__coerce__  # py2
__prepare__  # py3
__get__
__set__
__delete__
__enter__
__exit__
__contains__
__mising__
__reversed__
__int__
__float__
__complex__
__hex__
__oct__
__str__
__bytes__  # py3
__format__
__class__  # really strong proxy
```

Here was the class I was talking about: https://github.com/llllllllll/lazy_python/blob/047db06b67c46feb3e3cf7c8b37326c77c8d0177/lazy/thunk.py#L87 This is a really old version; however, it does have a pure python implementation of these methods. I have since switched to using C to define thunk and friends.

Also, why is `getattr` pure?

what about this class?

```python
class C:
    def __getattr__(self, key):
        return rand()
````

Also, the restriction on no methods that start with `_` disqualifies `collections.namedtuple` types because their public api methods start with `_`
